elastic/eui
# [Emotion] Convert EuiFormLayoutControl icons (3/3)

## Summary

Follow up to https://github.com/elastic/eui/pull/7954

As always, please [review by commit](https://github.com/elastic/eui/pull/7959/commits) - there's plenty of cleanup, some fixes around disabled styling, and a bunch of downstream snapshot changes.

## QA

The below links/steps should look the same as production:

- https://eui.elastic.co/pr_7959/storybook/?path=/story/forms-euiform-euiformcontrollayout--icon-shape&args=isInvalid:!true;isLoading:!true 
  - [x] Change `icon.side` to `left` and confirm that the :) face moves correctly
  - [x] Add an `onClick: true` key/value to the `icon` object and confirm that icon becomes a focusable button
  - [x] Click the `isDisabled` toggle and confirm that the :) face and clear button both correctly fade out and show a cursor indicating disabled status 
- https://eui.elastic.co/pr_7954/#/forms/form-controls#search-field
  - Type into the field (any text) and toggle the `loading` and `invalid` states
  - [x] Toggle the `compressed` state and confirm the clear button looks the same as production when small

### General checklist

- Browser QA
- Docs site QA - N/A
- Code quality checklist
    ~- [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**~
    - [x] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)** - _Ran with apparently no changes_
- Release checklist
    - [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately. - _Note that the component conversions are not listed in the changelog because they are not top-level public exports_
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A


<details><summary><b>Emotion checklist</b></summary>

**General**


---

**Unit tests**

- ~[ ] Converted Enzyme to RTL~

---

**Sass/Emotion conversion process**


---

**CSS tech debt**


---

**DOM Cleanup**


---

**Kibana due diligence**

  - ~[ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~
- [x] Any direct className usages that will need to be refactored (e.g. someone calling the `euiBadge` class on a div instead of simply using the `EuiBadge` component)~ None, thankfully

---

**Extras/nice-to-have**

</details>

Removed modifier classes:
- `.euiFormControlLayoutIcons--absolute`, `.euiFormControlLayoutIcons--static`, `.euiFormControlLayoutIcons--left`, `.euiFormControlLayoutIcons--right`
- `.euiFormControlLayoutCustomIcon--clickable`
- `.euiFormControlLayoutClearButton--small`

Biggest direct usage TODO to update:
- https://github.com/elastic/kibana/blob/f21d6e436e9f370b741bb495719af7e0f6ce36c1/src/plugins/unified_search/public/query_string_input/query_string_input.tsx#L850-L852
